program/src/processor.rs

```diff
@@ -325,28 +369,33 @@ impl Processor {
     }
 
     fn process_delegate(accounts: &[AccountInfo]) -> ProgramResult {
+        let signers = collect_signers(accounts);
         let account_info_iter = &mut accounts.iter();
+
+        // native asserts: 5 accounts (2 sysvars + 1 arbitrary)
         let stake_account_info = next_account_info(account_info_iter)?;
         let vote_account_info = next_account_info(account_info_iter)?;
         let clock_info = next_account_info(account_info_iter)?;
-        let clock = &Clock::from_account_info(clock_info)?;
         let _stake_history_info = next_account_info(account_info_iter)?;
-        let _stake_config_info = next_account_info(account_info_iter)?;
-        let stake_authority_info = next_account_info(account_info_iter)?;
+        let _arbitrary_info = next_account_info(account_info_iter)?;
 
-        let stake_history = &StakeHistorySysvar(clock.epoch);
+        // other accounts
+        // note this pattern is highly irregular. we would like stake authority to be fifth
+        // however, the fifth account used to be stake config, and it was removed without changing the assert
+        // there may be existing workflows that place stake authority fifth, or sixth, or omit it and self-sign
+        // we preserve the fifth unwrap above to avoid making this processor even more permissive
+        // let _stake_authority_info = next_account_info(account_info_iter);
 
-        // NOTE the existing program behaves as if this were false
-        // it should not break compat to check here, but may change errors
-        let (signers, _) = collect_signers(&[stake_authority_info], None, true)?;
+        let clock = &Clock::from_account_info(clock_info)?;
+        let stake_history = &StakeHistorySysvar(clock.epoch);
```

Comment:
im going to do a pr after this one to implement a version of `next_account_info()` which skips sysvars and stake config. this should be a perf improvement for 100% of transactions, once people omit the accounts, because getting sysvars by sycall is cheaper than building the `AccountInfo` in `entrypoint!`. once this program is live we can change the instruction builders to not include sysvars and backport all the way to 2.0 no matter what version is on mainnet

once i make that change, this code becomes:
* return error if fewer than five accounts
* collect signers
* create accounts iter
* unwrap four accounts